test/test_extension_point.py

```diff
@@ -139,6 +141,45 @@ def test_extension_point_blocklist():
         assert load.call_count == 0
 
 
+def test_extension_point_blocklist_override():
+    with patch.object(EntryPoint, 'load', return_value=None) as load:
+        clear_entry_point_cache()
+
+        my_extension_blocklist = EnvironmentVariable(
+            'MY_EXTENSION_BLOCKLIST', 'Foo bar baz')
+        override_blocklist_variable(my_extension_blocklist)
+
+        try:
+            # entry point in default blocklist variable can be loaded
+            load.reset_mock()
+            with EnvironmentContext(COLCON_EXTENSION_BLOCKLIST='group1'):
+                clear_entry_point_cache()
+                load_extension_point('extA', 'eA', 'group1')
+            assert load.call_count == 1
+
+            # entry point in custom blocklist variable can't be loaded
+            load.reset_mock()
+            with EnvironmentContext(MY_EXTENSION_BLOCKLIST='group1'):
+                clear_entry_point_cache()
+                with pytest.raises(RuntimeError) as e:
+                    load_extension_point('extA', 'eA', 'group1')
+                assert 'The entry point group name is listed in the ' \
+                    'environment variable' in str(e.value)
+            assert load.call_count == 0
+        finally:
+            override_blocklist_variable(None)
```

Comment:
This is what "resets" the blocklist variable and is responsible for the expected state difference between the test on https://github.com/colcon/colcon-core/pull/651/files#diff-307e796995136c87cf856ffb9123037838b3cc0a0e20838e2346f7f2a59784e0R155-R158 and the test on https://github.com/colcon/colcon-core/pull/651/files#diff-307e796995136c87cf856ffb9123037838b3cc0a0e20838e2346f7f2a59784e0R174-R180 correct?

Reply:
Precisely. First 2 checks makes sure that the override actually switched what variable the extension point subsystem is looking at, and the last check makes sure that the "reset to default" behavior works as expected. Also, because this changes globals, we don't want this test run to mess up other tests, so best to return it to the defaults.

I wanted to use a patch to "sandbox" this test a little more, but I'm pretty sure that the override will simply replace the mock.